transport-classes-kqueue/src/main/java/io/netty/channel/kqueue/KQueueIoHandler.java

```diff
@@ -266,10 +269,27 @@ public int run(IoHandlerContext context) {
             throw e;
         } catch (Throwable t) {
             handleLoopException(t);
+        } finally {
+            processCancelledRegistrations();
         }
         return handled;
     }
 
+    // Process all previous cannceld registrations and remove them from the registration map.
+    private void processCancelledRegistrations() {
+        for (;;) {
+            DefaultKqueueIoRegistration cancelledRegistration = cancelledRegistrations.poll();
+            if (cancelledRegistration == null) {
+                return;
+            }
```

Comment:
Can also be written like this

```suggestion
        DefaultKqueueIoRegistration cancelledRegistration;
        while ((cancelledRegistration = cancelledRegistrations.poll()) != null) {
```

Reply:
I actually prefer the other style and it's also what we use in other places.